### PR TITLE
refactor input variables to align with other domino terraform modules

### PR DESCRIPTION
**Pull request by steved**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -32,94 +44,105 @@ variable "allowed_ssh_ranges" {
...
+  default = {}
 }
 
-variable "google_dns_managed_zone" {
+variable "managed_dns" {
+  description = <<EOF
+  managed_dns = {
+    enabled = Whether to create DNS records in the given zone
+    name = Managed zone to modify
+    dns_name = DNS record names to create
```
Comment (fraenkel) [inline variables.tf:77]: names? or name?

File: variables.tf [hunk 2]
```diff
@@ -32,94 +44,105 @@ variable "allowed_ssh_ranges" {
...
+  EOF
 
-variable "location" {
-  type        = string
-  default     = "us-west1-b"
-  description = "The location (region or zone) of the cluster. A zone creates a single master. Specifying a region creates replicated masters accross all zones"
-}
+  type = object({
+    k8s_version     = optional(string, ""),
+    release_channel = optional(string, "RAPID"),
```
Comment (fraenkel) [inline variables.tf:126]: so we are changing the default? my only concern is that we would end up with 1.29 before we expected to support it.

File: variables.tf [hunk 3]
```diff
@@ -32,94 +44,105 @@ variable "allowed_ssh_ranges" {
...
-  }))
-  default = [
-    {
-      cidr_block   = "0.0.0.0/0"
-      display_name = "global-access"
-    }
-  ]
-  description = "Configuration options for master authorized networks. Default is for debugging only, and should be removed for production."
+  validation {
+    condition     = !var.gke.advanced_datapath || (var.gke.advanced_datapath && !var.gke.network_policies)
```
Comment (fraenkel) [inline variables.tf:143]: this doesn't match your comment
`Enable network policy switch. Has no effect when enable_advanced_datapath is true`

**Review by fraenkel — APPROVED** [review]
I prefer this style. Its much easier to add and grasp versus random single variables.

**Review by miguelhar — APPROVED** [review]

File: variables.tf (hunk 2 above)
Comment (steved) [reply variables.tf:126]: yeah, I thought this aligned better with what we're using internally. should I maybe set `k8s_version`?

Comment (fraenkel) [reply variables.tf:126]: well, you need both. the channel restricts the version.

File: variables.tf [hunk 4]
```diff
@@ -32,88 +44,105 @@ variable "allowed_ssh_ranges" {
...
+    public_access = optional(object({
+      enabled = optional(bool, false),
+      cidrs   = optional(list(string), [])
+    }), {}),
+    control_plane_ports      = optional(list(string), [])
+    advanced_datapath        = optional(bool, true),
+    network_policies         = optional(bool, false),
+    vertical_pod_autoscaling = optional(bool, true),
+    kubeconfig = optional(object({
+      path = optional(string, null)
```
Comment (fraenkel) [inline variables.tf:136]: what else would you expect here?

Comment (steved) [reply variables.tf:136]: it matches [EKS(https://github.com/dominodatalab/terraform-aws-eks/blob/main/modules/eks/variables.tf#L127), which has an `extra_args`

**Review by fraenkel — APPROVED** [review]
is there a point to being specific? we keep missing changes.  Should we open it all and just allow specifics if you really care?

**Comment by steved** [comment]
> is there a point to being specific? we keep missing changes. Should we open it all and just allow specifics if you really care?

I'm ok with this. @Secretions or @miguelhar any concerns?

**Comment by fraenkel** [comment]
The alternative is to have a set of known webhook ports that one should be using rather than picking your favorite port just because.
Some of these ports are concerning since it sounds like no HTTPS which will have to be resolved eventually.

**Comment by miguelhar** [comment]
> > is there a point to being specific? we keep missing changes. Should we open it all and just allow specifics if you really care?
> 
> I'm ok with this. @Secretions or @miguelhar any concerns?

Given previous experience in the realm of `if the module is public we are responsible for its usage` I rather have it restricted rather than `I got hacked because I used your module`

**Pull request merged**